### PR TITLE
[autogluon][build] AutoGluon 0.6.1 training containers

### PR DESCRIPTION
**Pull request by gradientsky**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by kace — DISMISSED** [review]

**Review by ohadkatz — APPROVED** [review]
LGTM! :shipit:

File: autogluon/training/docker/0.6/py3/Dockerfile.cpu.os_scan_allowlist.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{
+  "torch": [
```
Comment (saimidu) [inline autogluon/training/docker/0.6/py3/Dockerfile.cpu.os_scan_allowlist.json:2]: I believe this exception is only applicable to the AWS PT 1.12.1 whl, not the PT 1.12.1 whl on pypi or conda.

Please double check that this CVE is actually not applicable to the PT whl installed in this AG 0.6 DLC image.

File: autogluon/training/docker/0.6/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.pythonhosted.org pip \
+ && pip install --no-cache-dir -U setuptools wheel \
+ && pip install --no-cache-dir -U numpy numba \
+ && pip install --no-cache-dir -U certifi \
+    torch==${TORCH_VER} torchvision==${TORCH_VISION_VER} torchtext==${TORCH_TEXT_VER} --extra-index-url https://download.pytorch.org/whl/cpu \
+ && pip uninstall -y horovod \
+ && HOROVOD_WITH_PYTORCH=1 pip install --no-cache-dir horovod==0.24.3 \
+ && pip install --no-cache-dir -U autogluon==${AUTOGLUON_VERSION} \
+ && mim install -q mmcv-full \
+ && pip install --no-cache-dir -U mmdet
```
Comment (saimidu) [inline autogluon/training/docker/0.6/py3/Dockerfile.cpu:42]: @gradientsky Would you like to add integration tests for mmcv-full and mmdet? Are integration tests needed here for these newly added features?

Comment (gradientsky) [reply autogluon/training/docker/0.6/py3/Dockerfile.cpu:42]: These are transparent for the users when they use CV models, it's a backend change (similar how the transition from mxnet-based NNs to pytorch happened). mim is a model zoo for object detection models.

File: autogluon/training/docker/0.6/py3/Dockerfile.cpu [hunk 3]
```diff
@@ -0,0 +1,61 @@
+ARG PYTHON_VERSION=3.8.10
+
+FROM 763104351884.dkr.ecr.us-west-2.amazonaws.com/mxnet-training:1.9.0-cpu-py38-ubuntu20.04-sagemaker
+
+# PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK - see https://github.com/autogluon/autogluon/issues/2534
```
Comment (saimidu) [inline autogluon/training/docker/0.6/py3/Dockerfile.cpu:5]: Thanks for adding this explanatory comment! This is very helpful to have in the dockerfile!

File: autogluon/training/docker/0.6/py3/Dockerfile.cpu (hunk 2 above)
Comment (saimidu) [reply autogluon/training/docker/0.6/py3/Dockerfile.cpu:42]: Thanks! Resolving conversation on this comment.

File: autogluon/training/docker/0.6/py3/Dockerfile.cpu.os_scan_allowlist.json (hunk 1 above)
Comment (gradientsky) [reply autogluon/training/docker/0.6/py3/Dockerfile.cpu.os_scan_allowlist.json:2]: Addressed the exception via switching to PT base images built by AWS.

**Review by saimidu — DISMISSED** [review]
LGTM. Please revert dev config toml file changes.

**Review by kace — APPROVED** [review]

**Review by saimidu — APPROVED** [review]

**Pull request merged**